tests/testthat/test-problems.R

```diff
@@ -150,13 +150,20 @@ test_that("problems return the proper row number", {
 test_that("can promote vroom parse warning to error", {
   make_warning <- function() {
     x <- vroom(
-      I("a,b\n1.0,x\n"),
+      I("a\nx\n"),
       delim = ",",
-      col_types = "dd"
+      col_types = "d",
+      altrep = TRUE
     )
 
-    # Warning happens at print time so force a print
-    print(x)
+    # Trigger vroom parse warning while inside R's internal C code for `[` and ensure it doesn't crash R.
+    # `[` -> R's C function `do_subset()` -> ALTREP calls `vroom::real_Elt()` -> `vroom::warn_for_errors()`
+    # To avoid calling `cpp11::unwind_protect()` (which throws on longjmp, i.e. on `abort()`) while inside
+    # R's internal C code (which doesn't catch C++ exceptions), `vroom::warn_for_errors()` warns
+    # with base R's machinery rather than cpp11's
```

Comment:
> `vroom::warn_for_errors()` warns with base R's machinery rather than cpp11's

This review prompted me to read :point_up_2: for the first time in a long time and then compare it to the source for `warn_for_error()`:

https://github.com/tidyverse/vroom/blob/f21f3143f659745b06452ec8235819ca098fb2b0/src/vroom_errors.h#L85-L103

With fresh eyes, I feel like we can refine this to be even more precise. I'd say we're warning with cli and we're using base R machinery to access the relevant R function, as opposed to accessing it via cpp11 way.